From a77b36fd244d02e002d10fab84e97855b13c5774 Mon Sep 17 00:00:00 2001
From: Lina Iyer <lina.iyer@linaro.org>
Date: Tue, 28 Apr 2015 16:24:20 -0600
Subject: [PATCH 03/11] arm64: dts: Add power domain device bindings for
 MSM8916

Define L2 SAW as the power domain provider and individual cpus are the
power domain consumers.

Signed-off-by: Lina Iyer <lina.iyer@linaro.org>
---
 arch/arm64/boot/dts/qcom/msm8916.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/msm8916.dtsi b/arch/arm64/boot/dts/qcom/msm8916.dtsi
index f846135..cdc9d4a 100644
--- a/arch/arm64/boot/dts/qcom/msm8916.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8916.dtsi
@@ -53,6 +53,7 @@
 			next-level-cache = <&L2_0>;
 			qcom,saw = <&saw0>;
 			cpu-idle-states = <&CPU_SPC>;
+			power-domains = <&saw_l2>;
 		};
 
 		CPU1: cpu@1 {
@@ -64,6 +65,7 @@
 			next-level-cache = <&L2_0>;
 			qcom,saw = <&saw1>;
 			cpu-idle-states = <&CPU_SPC>;
+			power-domains = <&saw_l2>;
 		};
 
 		CPU2: cpu@2 {
@@ -75,6 +77,7 @@
 			next-level-cache = <&L2_0>;
 			qcom,saw = <&saw2>;
 			cpu-idle-states = <&CPU_SPC>;
+			power-domains = <&saw_l2>;
 		};
 
 		CPU3: cpu@3 {
@@ -86,6 +89,7 @@
 			next-level-cache = <&L2_0>;
 			qcom,saw = <&saw3>;
 			cpu-idle-states = <&CPU_SPC>;
+			power-domains = <&saw_l2>;
 		};
 
 		idle-states {
@@ -299,6 +303,7 @@
 		saw_l2: power-controller@B012000 {
 			compatible = "qcom,msm8916-saw2-v3.0-l2";
 			reg = <0xB012000 0x1000>, <0xB009000 0x1000>;
+			#power-domain-cells = <0>;
 		};
 
 		tcsr_mutex_regs: syscon@1905000 {
-- 
1.9.1

